### PR TITLE
upgrade dependencies and set golang version 1.18

### PR DESCRIPTION
**Pull request by jptosso**
- Upgrading golang to 1.18 might break some integrations
- This fixes the current CI problem

**Make sure that you've checked the boxes below before you submit PR:**

- [x] My code includes positive and negative tests.
- [x] I have an appropriate description with correct grammar.
- [x] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR :heart:

### CODE REVIEW

**Review by piyushroshan — APPROVED** [review]

File: .github/workflows/go-ftw.yml [hunk 1]
```diff
@@ -39,20 +39,21 @@ jobs:
...
         mv main.go caddy/main.go
-        go get github.com/jptosso/coraza-caddy@ffa1472
+        go get github.com/corazawaf/coraza-caddy@8ed5dd2
         go mod tidy
         go get ./...
         go get -u github.com/jptosso/coraza-pcre@593842e
-        go install caddy/main.go
-        sudo setcap CAP_NET_BIND_SERVICE=+eip $(which caddy)
+        go build -o caddybin caddy/main.go
+        file caddybin
```
Comment (anuraaga) [inline .github/workflows/go-ftw.yml:52]:
```suggestion
```

Comment (anuraaga) [reply .github/workflows/go-ftw.yml:52]: Presume this was for debugging

Comment (jptosso) [reply .github/workflows/go-ftw.yml:52]: Yes, I actually think it's useful to keep some commands like ls and file in github actions for future debugging. There were cases where the caddybin file wasn't created, so it's good to check for it

**Review by anuraaga — APPROVED** [review]

**Review by bxlxx — APPROVED** [review]
LGTM

**Pull request merged**